src/circuit/freeze.rs

```diff
@@ -180,9 +180,9 @@ pub(crate) struct FreezeWitnessVar {
 
 impl FreezeWitnessVar {
```

Comment:
a general question: do you think it is better to bind the `FreezeWitnessVar` with `C` or its individual functions? I don't have a strong preference -- Variable itself should not be bounded to `C` but here it is a Cap specific variable...

Reply:
These `XXXVar` are not generic over `<C: CapConfig>` because all constituent fields are normally just `Variable`, `PointVariable`. 
I don't see a strong need to add an unconstrained generic param (even tho we can with the help of `PhantomData`)

Reply:
I wasn't appreciating this question enough last night. I change my mind. 

> do you think it is better to bind the FreezeWitnessVar with C or its individual functions?

I now tend to think we should bind with `FreezeWitnessVar<C>` since such variable is more meaningful under a `CapConfig`.